src/StableBuffer.mo

```diff
@@ -17,62 +17,190 @@
 /// determined at construction and cannot be changed).
 
 import Prim "mo:⛔";
+import Result "mo:base/Result";
+import Order "mo:base/Order";
+import Array "mo:base/Array";
 
 module {
+  type Order = Order.Order;
+
+  // The following constants are used to manage the capacity.
+  // The length of `elements` is increased by `INCREASE_FACTOR` when capacity is reached.
+  // The length of `elements` is decreased by `DECREASE_FACTOR` when capacity is strictly less than
+  // `DECREASE_THRESHOLD`.
+
+  // INCREASE_FACTOR = INCREASE_FACTOR_NUME / INCREASE_FACTOR_DENOM (with floating point division)
+  // Keep INCREASE_FACTOR low to minimize cycle limit problem
+  private let INCREASE_FACTOR_NUME = 3;
+  private let INCREASE_FACTOR_DENOM = 2;
   private let DECREASE_THRESHOLD = 4; // Don't decrease capacity too early to avoid thrashing
   private let DECREASE_FACTOR = 2;
+  private let DEFAULT_CAPACITY = 8;
+
+  private func newCapacity(oldCapacity : Nat) : Nat {
+    if (oldCapacity == 0) {
+      1;
+    } else {
+      // calculates ceil(oldCapacity * INCREASE_FACTOR) without floats
+      ((oldCapacity * INCREASE_FACTOR_NUME) + INCREASE_FACTOR_DENOM - 1) / INCREASE_FACTOR_DENOM;
+    };
+  };
 
   public type StableBuffer<X> = {
-    initCapacity : Nat;
-    var count : Nat;
-    var elems : [var ?X];
+    var _size : Nat; // avoid name clash with `size()` method
+    var elements : [var ?X];
```

Comment:
This is a complete breaking change with the `StableBuffer` type. Since this is now stable, the inner properties are expected to be the same name before & after upgrade.

Can you revert this to keep the original property names?

This applies to everywhere else that `initCapacity`, `count`, and `elems` are referenced in the PR.

Reply:
I guess the alternative is that we release a `StableBufferV2` repo, which is recommended as it is more in sync with the one in base. That actually might be easier, and then we don't break people.